### PR TITLE
support optional fields for AverageEnvInfoMetric

### PR DESCRIPTION
**Pull request by hnyu**
Sometimes we have lots of stuff in `time_step.env_info` but we don't want to metric everyone of them.

### CODE REVIEW

File: alf/metrics/metrics.py [hunk 1]
```diff
@@ -553,6 +555,7 @@ def _extract_metric_values(self, time_step):
                            torch.ones_like(time_step.step_type))
 
 
+@alf.configurable(whitelist=['fields'])
```
Comment (breakds) [inline alf/metrics/metrics.py:558]: Suggest `whitelist_fields` so that the name indicates this is a whitelist.

**Review by breakds — COMMENTED** [review]
Thanks Haonan!

**Review by breakds — APPROVED** [review]

**Comment by breakds** [comment]
Works as intended and pretty well. I'll merge it.
